bigtable-client-core/src/main/java/com/google/cloud/bigtable/grpc/BigtableDataGrpcClient.java

```diff
@@ -238,28 +240,67 @@ public Row readModifyWriteRow(ReadModifyWriteRowRequest request) {
 
   @Override
   public ResultScanner<Row> readRows(ReadRowsRequest request) {
-    return readRows(request, retryOptions.enableRetries());
+    boolean resumable = hasAdaptedWhileMatchFilter(request.getFilter());
+    return readRows(request, retryOptions.enableRetries(), resumable);
   }
 
   /**
-   * Begin reading rows, optionally with a resumable scanner.
+   * Recursively checks to see if the filter has adapted a {@link WhileMatchFilter}. If yes, a
+   * resumable scanner needs to be used. A label transformer is used when adapting a {@link
+   * WhileMatchFilter}.
+   *
+   * @param filter to be checked
+   * @return true if the filter has adapted {@link WhileMatchFilter}
    */
-  private ResultScanner<Row> readRows(ReadRowsRequest request, boolean resumable) {
-    // Delegate all resumable operations to the scanner. It will request a non-resumable
-    // scanner during operation.
-    if (resumable) {
-      return new ResumingStreamingResultScanner(
-        retryOptions,
-          request,
-          new BigtableResultScannerFactory() {
+  @SuppressWarnings("javadoc")
+  private static boolean hasAdaptedWhileMatchFilter(RowFilter filter) {
```

Comment:
Drive-by code review comment: I'd argue that this class / package shouldn't have any notion of HBase concepts like WhileMatchFilters. Especially since label transformers could be applied by users of a lightweight client (this thin wrapper over grpc).